### ISSUE TITLE
Support recent search tracking with Room

### ISSUE THREAD

**Issue by gouri-panda** [kiwix/kiwix-android#3111]
Should be done:
* A kind of generic and abstract accessor to access Room data
* Secure the Room DB is created properly at start (if it is not easy to create on the fly) if necessary (first installation or migration context)
* The data structure and accessor to the recent searches (based on what exists in ObjectBox)
* Link with the existing code around the search UI (disconnect this from ObjectBox)
* Write the automated tests (Unit tests and maybe one UI test) for that new Room code (adapt what exists for ObjectBox)
* Write the migration function (for the data) from the ObjectBox recent_searches DB to the Room DB
* Write automated tests for that migration function

Once this is done #3115 will have to be implemented to secure we can test the migration.

RecentSearchEntity (Responisble for recent searches) https://github.com/kiwix/kiwix-android/blob/e2c901bf22034ea891c29ce12bc5793df7eccf52/core/src/main/java/org/kiwix/kiwixmobile/core/dao/entities/RecentSearchEntity.kt

**Comment by kelson42** [comment]
This is not handled by libkiwix, so you can go ahead with this first.

**Comment by gouri-panda** [comment]
Ok @kelson42

**Comment by kelson42** [comment]
@gouri-panda Any PR ready for this first ticket?

**Comment by gouri-panda** [comment]
@kelson42 Sorry! I forgot to mention it in #3132 . In #3132 first 2 steps has already been done :)

**Comment by kelson42** [comment]
@gouri-panda Please open a PR fixing this and nothing else. I have explained in length why. I will close #3132 too for the very same reasons. If you really can't develop the features step by step, which seems the case here, proceed like you want in your branch; but regarding PR, we will do step by step and there won't be a "big-bang".

**Comment by gouri-panda** [comment]
@kelson42 Ok. I'll first migrate bookmarks, library, and recent search to Room. I'll change some in #3132 .

**Comment by kelson42** [comment]
No, this ticket is only about recent searches.

**Comment by gouri-panda** [comment]
No in the #3132.

**Comment by kelson42** [comment]
One PR for one ticket, so "yes" you have to open a new PR.

**Comment by kelson42** [comment]
@MohitMaliFtechiz The migration of recent search was the first step for room migration. This step is like a template for the other dataset to migrate, has nothing to do with libkiwix and if Incorrectly remember @gouri-panda was almost over with it after a few back and forth. can you please read all the PR and remark and based on his work prepare anew PR?
